crates/xilem_html/src/diff.rs

```diff
@@ -0,0 +1,61 @@
+use std::{cmp::Ordering, collections::BTreeMap, iter::Peekable};
+
+pub fn diff_tree_maps<'a, K: Ord, V: PartialEq>(
```

Comment:
The library doesn't have much (any?) test coverage ATM, but this would be a good candidate for some tests checking the diff sequence. Not a blocker tho - we can add them later.

Reply:
Yeah, probably a good idea, I can add them here.